### PR TITLE
#166789880 Users should be able to comment on articles

### PR DESCRIPTION
**Pull request by manzif**
#### What does this PR do?

 - Users should be able to comment on articles

#### Description of Task to be completed?

  - create comments table
  - add comment on an article
  - delete comment
  - get all comments

#### How should this be manually tested?

- When you are a registered user, you can comment on any article, you can also delete your own comments. But everyone can see all the comments of any article.

#### Any background context you want to provide?

- N/A

#### What are the relevant pivotal tracker stories?

- [#166789880](https://www.pivotaltracker.com/story/show/166789880)

## 1. Create a new comment

image

image

image

image

   ## 2. delete a  comment

image

image

   ## 3. get all  comment

image

#### Questions:

- N/A

### CODE REVIEW

**Review by Dom58 — COMMENTED** [review]
squash commits

**Review by Emile-Nsengimana — COMMENTED** [review]
there is still some hound issues

File: src/routes/api/comments.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import express from 'express';
+import commentController from '../../controllers/commentController';
+import isAuth from '../../middlewares/auth';
+import ValidationComments from '../../helpers/commentValidation';
```
Comment (Emile-Nsengimana) [inline src/routes/api/comments.js]: use camel case

File: swagger.json [hunk 2]
```diff
@@ -5,7 +5,7 @@
         "version": "1.0.0",
         "description": "Authors Haven Documentation with Swagger"
     },
-    "host": "ah-lobos-backend-swagger.herokuapp.com",
+    "host": "localhost:7000",
```
Comment (Emile-Nsengimana) [inline swagger.json]: remove localhost

Comment (manzif) [reply swagger.json]: Resolved

File: src/routes/api/comments.js (hunk 1 above)
Comment (manzif) [reply src/routes/api/comments.js]: Resolved

**Review by adafia — APPROVED** [review]

**Review by Emile-Nsengimana — APPROVED** [review]

**Review by Dom58 — APPROVED** [review]

**Comment by Emile-Nsengimana** [comment]
add mockup image(s) on the PR

File: src/controllers/commentController.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+
+dotenv.config();
+
+const { Comments, Articles } = model;
+
+/**
+ * comment controller
+ */
+class commentManager {
```
Comment (emukungu) [inline src/controllers/commentController.js]: use Pascal naming for classes.

**Review by emukungu — CHANGES_REQUESTED** [review]

File: src/controllers/commentController.js (hunk 3 above)
Comment (manzif) [reply src/controllers/commentController.js]: I resolved it

**Comment by emukungu** [comment]
improve you commit header

**Comment by manzif** [comment]
> improve you commit header

I improved it

**Pull request merged**